src/main/java/org/broadinstitute/hellbender/tools/spark/bwa/BwaSpark.java

```diff
@@ -40,15 +47,25 @@ public boolean requiresReads() {
     }
 
     @Override
-    protected void runTool(final JavaSparkContext ctx) {
+    public List<ReadFilter> getDefaultReadFilters() {
+        // 1) unmapped or neither secondary nor supplementary and 2) has some sequence
+        return Collections.singletonList(ReadFilterLibrary.PRIMARY_ALIGNMENT.and(ReadFilterLibrary.SEQ_IS_STORED));
```

Comment:
I think it's preferred to have the filters be separate entries in a list rather than chaining them.  If they're separated out individual then they can be toggled on/off on the command line and will show up in help output appropriately.  If they're chained into a single filter then I don't think they can be inspected properly.